### PR TITLE
Introducing integration tests to Sahimi

### PR DESCRIPTION
**Pull request by johnsimons**
This PR introduces the ability to write integration tests that start at the `IActionHandler` and execute Calamari commands and back again.

Calamari is run in the same process, so F5 friendly 😉

### CODE REVIEW

File: source/Sashimi.Terraform.Tests/AllowPluginDownloadsShouldBeDisabled.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using NUnit.Framework;
+using Sashimi.Server.Contracts.ActionHandlers;
+using Sashimi.Terraform.ActionHandler;
+using Sashimi.Tests.Shared.Extensions;
+using Sashimi.Tests.Shared.Server;
+
+namespace Sashimi.Terraform.Tests
+{
+    [TestFixture]
+    public class AllowPluginDownloadsShouldBeDisabled : BaseTest
```
Comment (flin-8) [inline source/Sashimi.Terraform.Tests/AllowPluginDownloadsShouldBeDisabled.cs:11]: Is this class needed?

File: source/Sashimi.Terraform.Tests/TerraformTests.cs [hunk 2]
```diff
@@ -0,0 +1,474 @@
...
+using Sashimi.Server.Contracts;
+using Sashimi.Server.Contracts.ActionHandlers;
+using Sashimi.Terraform.ActionHandler;
+using Sashimi.Tests.Shared;
+using Sashimi.Tests.Shared.Server;
+
+namespace Sashimi.Terraform.Tests
+{
+    [TestFixture]
+    public class TerraformTests : BaseTest
```
Comment (flin-8) [inline source/Sashimi.Terraform.Tests/TerraformTests.cs:27]: The naming convention is to use `Fixture`?

File: source/Sashimi.Tests.Shared/Sashimi.Tests.Shared.csproj [hunk 3]
```diff
@@ -13,7 +13,10 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <ProjectReference Include="..\Calamari.Terraform\Calamari.Terraform.csproj" />
```
Comment (flin-8) [inline source/Sashimi.Tests.Shared/Sashimi.Tests.Shared.csproj:16]: Shouldn't need the `Terraform` reference

Comment (johnsimons) [reply source/Sashimi.Tests.Shared/Sashimi.Tests.Shared.csproj]: Fixed

File: source/Sashimi.Terraform.Tests/TerraformTests.cs (hunk 2 above)
Comment (johnsimons) [reply source/Sashimi.Terraform.Tests/TerraformTests.cs]: Fixed

File: source/Sashimi.Terraform.Tests/AllowPluginDownloadsShouldBeDisabled.cs (hunk 1 above)
Comment (johnsimons) [reply source/Sashimi.Terraform.Tests/AllowPluginDownloadsShouldBeDisabled.cs]: Nope, removed

**Pull request merged**